test-suite-ex/shared/src/test/scala/org/scalajs/testsuite/javalib/util/FormatterTestEx.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util
+
+import java.util.{Formatter, Locale}
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.testsuite.utils.Platform.executingInJVM
+
+/** Additional tests for java.lang.String that require `java.util.Locale`
+ *  as well as classes in `java.text.*`.
+ */
+class FormatterTestEx {
+
+  /* Note: there is no test for a grouping size != 3, because I (sjrd) could
+   * not find any locale for which it would be different from 3.
+   */
+
+  val French = new Locale("fr")           // decimal sep ','  grouping sep '\u00A0'
+  val HindiIndia = new Locale("hi", "IN") // non-ASCII digits
+  val Turkish = new Locale("tr")          // special uppercase behavior
+
+  final val Nbsp = '\u00A0' // non-breaking space
+
+  def assertF(locale: Locale, expected: String, format: String, args: Any*): Unit = {
+    // Locale passed as constructor parameter
+    val fmt1 = new Formatter(locale)
+    val res1 = fmt1.format(format, args.asInstanceOf[Seq[AnyRef]]: _*).toString()
+    fmt1.close()
+    assertEquals(expected, res1)
+
+    // Locale passed as argument to `format`
+    val fmt2 = new Formatter()
+    val res2 = fmt2.format(locale, format, args.asInstanceOf[Seq[AnyRef]]: _*).toString()
+    fmt2.close()
+    assertEquals(expected, res2)
+  }
+
+  @Test def testLocale(): Unit = {
+    assertEquals(Locale.ROOT, new Formatter().locale())
+
+    assertSame(Locale.ROOT, new Formatter(Locale.ROOT).locale())
+    assertSame(French, new Formatter(French).locale())
+    assertSame(null, new Formatter(null: Locale).locale())
+
+    // Calling `format` with an explicit locale does not change the `locale()`
+    val formatter = new Formatter(French)
+    formatter.format(HindiIndia, "")
+    assertSame(French, formatter.locale())
+  }
+
+  @Test def testFormatFrench(): Unit = {
+    assertF(French, "1\u00A0234\u00A0567", "%,d", 1234567)
+    assertF(French, "1\u00A0234\u00A0567,89", "%,.2f", 1234567.89)
+    assertF(French, "0012", "%04d", 12)
+  }
+
+  @Test def testFormatHindiIndia(): Unit = {
+    assertF(HindiIndia, "\u0967,\u0968\u0969\u096A,\u096B\u096C\u096D", "%,d", 1234567)
+    assertF(HindiIndia, "\u0967,\u0968\u0969\u096A,\u096B\u096C\u096D.\u096E\u096F", "%,.2f", 1234567.89)
+    assertF(HindiIndia, "\u0966\u0966\u0967\u0968", "%04d", 12)
```

Comment:
Consider writing these out in actual unicode. (I believe we have other funky unicode characters in the codebase anyways). Like this, it is very hard to check if the test is reasonable.